File: backend/app/models/geo_response.py
from sqlalchemy import Column, String, DateTime, Text, ForeignKey, Integer, Enum as SQLEnum
from sqlalchemy.sql import func
from app.core.database import Base
import uuid
import enum


class ResponseStatus(enum.Enum):
    """GEO response status"""
    SUCCESS = "success"
    FAILED = "failed"
    PARTIAL = "partial"


class GEOResponse(Base):
    """GEO AI responses - linked to prompts and entities"""
    __tablename__ = "geo_responses"
    
    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    geo_prompt_id = Column(String(36), ForeignKey("geo_prompts.id", ondelete="CASCADE"), nullable=False, index=True)
    business_profile_id = Column(String(36), ForeignKey("business_profiles.id", ondelete="CASCADE"), nullable=False, index=True)
    
    # Request data
    input_data = Column(Text)  # JSON - the actual data sent to LLM
    
    # Response data
    response_text = Column(Text)  # Raw LLM response
    parsed_response = Column(Text)  # JSON - structured response
    
    # Metadata
    status = Column(SQLEnum(ResponseStatus), default=ResponseStatus.SUCCESS, nullable=False)
    error_message = Column(Text)
    
    # Usage metrics
    tokens_used = Column(Integer)
    execution_time_ms = Column(Integer)
    
    # Timestamps
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    
    def __repr__(self):
        return f"<GEOResponse(id={self.id}, status={self.status})>"

File: backend/src/models/GeoResponse.ts
import { Schema, model, Document, Types } from 'mongoose';

export interface IGeoResponse extends Document {
  _id: Types.ObjectId;
  geoPromptId: Types.ObjectId;
  businessProfileId: Types.ObjectId;
  inputData?: string;
  responseText?: string;
  parsedResponse?: string;
  status: 'SUCCESS' | 'FAILED' | 'PARTIAL';
  errorMessage?: string;
  tokensUsed?: number;
  executionTimeMs?: number;
  createdAt: Date;
}

const geoResponseSchema = new Schema<IGeoResponse>(
  {
    geoPromptId: {
      type: Schema.Types.ObjectId,
      ref: 'GeoPrompt',
      required: true,
      index: true,
    },
    businessProfileId: {
      type: Schema.Types.ObjectId,
      ref: 'BusinessProfile',
      required: true,
      index: true,
    },
    inputData: String,
    responseText: String,
    parsedResponse: String,
    status: {
      type: String,
      enum: ['SUCCESS', 'FAILED', 'PARTIAL'],
      required: true,
    },
    errorMessage: String,
    tokensUsed: Number,
    executionTimeMs: Number,
  },
  {
    timestamps: true,
  }
);

export const GeoResponse = model<IGeoResponse>('GeoResponse', geoResponseSchema);
